src/infra/account/secureAccessKey/secureAccessKeyCmdRepo.go

```diff
@@ -0,0 +1,202 @@
+		accountQueryRepo:         accountInfra.NewAccountQueryRepo(persistentDbSvc),
+	}
+}
+
+func (repo *SecureAccessKeyCmdRepo) createSecureAccessKeysFileIfNotExists(
+	accountUsername valueObject.Username,
+) error {
+	accountUsernameStr := accountUsername.String()
+
+	secureAccessKeysDirPath := "/home/" + accountUsernameStr + "/.ssh"
+	if !infraHelper.FileExists(secureAccessKeysDirPath) {
+		err := infraHelper.MakeDir(secureAccessKeysDirPath)
+		if err != nil {
+			return errors.New("CreateSecureAccessKeysDirectoryError: " + err.Error())
+		}
+	}
+
+	secureAccessKeysFilePath := secureAccessKeysDirPath + "/authorized_keys"
+	if infraHelper.FileExists(secureAccessKeysFilePath) {
+		return nil
+	}
+
+	_, err := os.Create(secureAccessKeysFilePath)
+	if err != nil {
+		return errors.New("CreateSecureAccessKeysFileError: " + err.Error())
+	}
+
+	_, err = infraHelper.RunCmd(
+		"chown", "-R", accountUsernameStr, secureAccessKeysFilePath,
+	)
+	if err != nil {
+		return errors.New("ChownSecureAccessKeysFileError: " + err.Error())
+	}
+
+	return nil
+}
+
+func (repo *SecureAccessKeyCmdRepo) allowAccountSecureRemoteConnection(
+	accountId valueObject.AccountId,
+) error {
+	accountUsername, err := infraHelper.RunCmdWithSubShell(
+		"awk -F: '$3 == " + accountId.String() +
+			" && $7 != \"/bin/bash\" {print $1}' /etc/passwd",
+	)
+	if err != nil {
+		return errors.New("ReadUnixUsernameFromFileError: " + err.Error())
+	}
+
+	_, err = infraHelper.RunCmdWithSubShell(
+		"chsh -s /bin/bash " + accountUsername,
+	)
+	if err != nil {
+		return errors.New("ChangeDefaultBashError: " + err.Error())
+	}
+
+	return nil
+}
+
+func (repo *SecureAccessKeyCmdRepo) recreateSecureAccessKeysFile(
+	accountId valueObject.AccountId,
+	accountUsername valueObject.Username,
+) error {
+	keysFilePath := "/home/" + accountUsername.String() + "/.ssh/authorized_keys"
+	if !infraHelper.FileExists(keysFilePath) {
+		err := repo.createSecureAccessKeysFileIfNotExists(accountUsername)
+		if err != nil {
+			return err
+		}
+	}
+
+	readRequestDto := dto.ReadSecureAccessKeysRequest{
+		Pagination: dto.Pagination{
+			ItemsPerPage: 1000,
+		},
+		AccountId: accountId,
+	}
+	keys, err := repo.secureAccessKeyQueryRepo.Read(readRequestDto)
+	if err != nil {
+		return err
+	}
+
+	keysFileContent := ""
+	for _, key := range keys.SecureAccessKeys {
+		keysFileContent += key.Content.String() + " " + key.Name.String() + "\n"
+	}
+
+	shouldOverwrite := true
+	err = infraHelper.UpdateFile(keysFilePath, keysFileContent, shouldOverwrite)
+	if err != nil {
+		return errors.New("UpdateSecureAccessKeysFileContentError: " + err.Error())
+	}
+
+	return nil
+}
+
+func (repo *SecureAccessKeyCmdRepo) Create(
+	createDto dto.CreateSecureAccessKey,
+) (keyId valueObject.SecureAccessKeyId, err error) {
+	account, err := repo.accountQueryRepo.ReadById(createDto.AccountId)
+	if err != nil {
+		return keyId, errors.New("AccountNotFound")
+	}
+
+	err = repo.createSecureAccessKeysFileIfNotExists(account.Username)
```

Comment:
I would leave the "existence check" here on the main method and call the create file after it returns the key does not exist. It would be better than have such a big method name. However, that's just a suggestion, you do you.

Reply:
By the way, the naming of the method is misleading. You're not creating the security access keys file, you're creating the .authorized_keys file. That's different. This is probably something you should be creating on the installation of OpenSSH rather than here. So this method should go.

Reply:
I moved the entire infrastructure to the Account, so this method became a simple creation method called `createAuthorizedKeysFile`, which is executed within the user creation process. There are no more concerns about verbose names or functions being placed in inappropriate locations!